packages/terra-layout/src/Layout.jsx

```diff
@@ -0,0 +1,193 @@
+    const isFixedMenu = currentState.size === 'large' || currentState.size === 'huge';
+    const isHoverMenu = currentState.size === 'medium';
+    const isToggleMenu = currentState.size === 'tiny' || currentState.size === 'small';
+
+    const menuIsPresent = !!props.menu;
+
+    return Object.assign({}, currentState || {}, {
+      isHoverMenu,
+      isFixedMenu,
+      isToggleMenu,
+      menuIsPresent,
+      menuIsOpen: menuIsPresent && (currentState.menuIsOpen || isFixedMenu),
+      menuIsPinned: menuIsPresent && currentState.menuIsPinned && isHoverMenu,
+    });
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.toggleMenu = this.toggleMenu.bind(this);
+    this.togglePin = this.togglePin.bind(this);
+    this.updateSize = this.updateSize.bind(this);
+    this.renderHeader = this.renderHeader.bind(this);
+    this.renderMenu = this.renderMenu.bind(this);
+    this.renderContent = this.renderContent.bind(this);
+
+    this.state = Layout.stateForProps(props, {
+      size: getBreakpointSize(),
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.updateSize);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(Layout.stateForProps(nextProps, this.state));
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateSize);
+  }
+
+  updateSize() {
+    const newSize = getBreakpointSize();
+
+    if (this.state.size !== newSize) {
+      this.setState(Layout.stateForProps(this.props, {
+        size: newSize,
+      }));
+    }
+  }
+
+  toggleMenu() {
+    this.setState({
+      menuIsOpen: !this.state.menuIsOpen,
+    });
+  }
+
+  togglePin() {
+    this.setState({
+      menuIsPinned: !this.state.menuIsPinned,
+    });
+  }
+
+  renderHeader() {
+    const { header } = this.props;
+    const { size, menuIsOpen, isToggleMenu, menuIsPresent } = this.state;
+
+    if (!header) {
+      return null;
+    }
+
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    return React.cloneElement(header, {
+      layoutConfig: {
+        size,
+        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
+        menuIsOpen,
+      },
+    });
+  }
+
+  renderMenu() {
+    const { menu } = this.props;
+    const { size, menuIsOpen, menuIsPinned, isToggleMenu, isHoverMenu, menuIsPresent } = this.state;
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    if (!menuIsPresent) {
+      return null;
+    }
+
+    return React.cloneElement(menu, {
+      layoutConfig: {
+        size,
+        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
+        menuIsOpen,
+        togglePin: isHoverMenu ? this.togglePin : undefined,
+        menuIsPinned,
+      },
+    });
+  }
+
+  renderContent() {
+    const { children } = this.props;
+    const { size, menuIsOpen, isToggleMenu, menuIsPresent } = this.state;
+    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
+
+    if (!children) {
```

Comment:
Should the layout content be required?

Reply:
I don't think I want to make it required, but I do think I want to render the ContentContainer below even if the children are missing. So I'll still swizzle this around.